packages/presentation/lib/ui/splash_screen/bloc/splash_bloc.dart

```diff
@@ -1,33 +1,82 @@
+import 'dart:io';
+
 import 'package:domain/use_case/splash_duration_use_case.dart';
+import 'package:domain/use_case/check_version_use_case.dart';
 import 'package:presentation/base/bloc.dart';
+import 'package:presentation/generated/l10n.dart';
+import 'package:presentation/library/const/links.dart';
+import 'package:presentation/navigation/base_arguments.dart';
 import 'package:presentation/ui/movie_page/movie_widget.dart';
+import 'package:presentation/ui/splash_screen/bloc/splash_tile.dart';
+import 'package:url_launcher/url_launcher.dart';
 
-abstract class SplashBloc extends Bloc {
+abstract class SplashBloc extends Bloc<BaseArguments, SplashTile> {
   factory SplashBloc(
     SplashDurationUseCase durationUseCase,
+    CheckVersionUseCase versionUseCase,
   ) =>
       SplashBlocImpl(
         durationUseCase,
+        versionUseCase,
       );
+
+  Future<void> updateTap();
 }
 
-class SplashBlocImpl extends BlocImpl implements SplashBloc {
+class SplashBlocImpl extends BlocImpl<BaseArguments, SplashTile>
+    implements SplashBloc {
+  var _tile = SplashTile.init();
   final SplashDurationUseCase _durationUseCase;
+  final CheckVersionUseCase _versionUseCase;
 
   SplashBlocImpl(
     this._durationUseCase,
+    this._versionUseCase,
```

Comment:
seems like u forget to delete this._durationUseCase
cuz u don't need 3 sec delay anymore. Now _versionUseCase is a delay

Reply:
where did you get this from?) this is not specified in the task, but on the contrary, that the user must return to the stream, respectively, to the delay

Reply:
I think this supposed by task. earlier we used _durationUseCase like a plug. Now we do not need this plug on my opinion, cuz check version use case is our delay.

Reply:
but up to u =)

Reply:
I completely agree with you, but I still do it as indicated in the task)

Reply:
Add a delay for 3 sec (imitation of API call which will be added further) and redirect the user to Main Page (c) https://moneyman-jira.atlassian.net/browse/IO-493
"imitation of API call which will be added further" time stil  is not come?

Reply:
:D I agree, the future is here